cluster/centos/util.sh

```diff
@@ -218,25 +239,30 @@ echo "[INFO] tear-down-node on $1"
 #   ETCD_SERVERS
 #   SERVICE_CLUSTER_IP_RANGE
 function provision-master() {
-  echo "[INFO] Provision master on ${MASTER}"
-  local master_ip=${MASTER#*@}
-  ensure-setup-dir ${MASTER}
-
-  # scp -r ${SSH_OPTS} master config-default.sh copy-files.sh util.sh "${MASTER}:${KUBE_TEMP}"
-  kube-scp ${MASTER} "${ROOT}/../saltbase/salt/generate-cert/make-ca-cert.sh ${ROOT}/binaries/master ${ROOT}/master ${ROOT}/config-default.sh ${ROOT}/util.sh" "${KUBE_TEMP}"
-  kube-ssh "${MASTER}" " \
+  echo "[INFO] Provision master on $1"
+  local master=$1
+  local master_ip=${master#*@}
+  ensure-setup-dir ${master}
+
+  # scp -r ${SSH_OPTS} master config-default.sh copy-files.sh util.sh "${master}:${KUBE_TEMP}"
+  kube-scp ${master} "${ROOT}/ca-cert ${ROOT}/binaries/master ${ROOT}/master ${ROOT}/config-default.sh ${ROOT}/util.sh" "${KUBE_TEMP}"
+  kube-ssh "${master}" " \
     rm -rf /opt/kubernetes/bin; \
     sudo cp -r ${KUBE_TEMP}/master/bin /opt/kubernetes; \
+    sudo mkdir -p /srv/kubernetes; sudo cp -f ${KUBE_TEMP}/ca-cert/* /srv/kubernetes; \
     sudo chmod -R +x /opt/kubernetes/bin; \
     sudo ln -s /opt/kubernetes/bin/* /usr/local/bin/; \
-    sudo bash ${KUBE_TEMP}/make-ca-cert.sh ${master_ip} IP:${master_ip},IP:${SERVICE_CLUSTER_IP_RANGE%.*}.1,DNS:kubernetes,DNS:kubernetes.default,DNS:kubernetes.default.svc,DNS:kubernetes.default.svc.cluster.local; \
-    sudo bash ${KUBE_TEMP}/master/scripts/etcd.sh; \
+    sudo bash ${KUBE_TEMP}/master/scripts/etcd.sh ${master_ip} ${ETCD_DISCOVERY}; \
     sudo bash ${KUBE_TEMP}/master/scripts/apiserver.sh ${master_ip} ${ETCD_SERVERS} ${SERVICE_CLUSTER_IP_RANGE} ${ADMISSION_CONTROL}; \
-    sudo bash ${KUBE_TEMP}/master/scripts/controller-manager.sh ${master_ip}; \
-    sudo bash ${KUBE_TEMP}/master/scripts/flannel.sh ${ETCD_SERVERS} ${FLANNEL_NET}; \
-    sudo bash ${KUBE_TEMP}/master/scripts/scheduler.sh ${master_ip}"
+    sudo bash ${KUBE_TEMP}/master/scripts/controller-manager.sh ${MASTER_ADVERTISE_ADDRESS}; \
+    sudo bash ${KUBE_TEMP}/master/scripts/scheduler.sh ${MASTER_ADVERTISE_ADDRESS}"
 }
 
+function post-provision-master() {
```

Comment:
Please, document this function and add description of assumed values & parameters.